packages/pn-commons/src/components/CodeModal/CodeInput.tsx

```diff
@@ -26,6 +27,7 @@ type Props = {
  */
 const CodeInput = memo(
   forwardRef(({ initialValues, isReadOnly, hasError, onChange }: Props, ref) => {
+    const [currentValues, setCurrentValues] = useState(initialValues);
```

Comment:
Qui si stanno mischiando due approcci (ref + state) che potrebbero portare ad un rerendering eccessivo e non necessario del componente.
L'idea è quella di scegliere uno dei due. Le soluzioni sono due:
- si mantiene il ref per il focus sugli input e si usa lo state per il value
- si mantiene il ref sia per il focus che per il value

Reply:
Assolutamente d'accordo, l'idea era di fare un fix rapido per risolvere il problema preoccupandosi del refactoring e dell'ottimizzazione una volta risolto il bug di mancato aggiornamento. Impostando uno stato per il valore del componente, aggiornando questo stato nella changeHandler e impostando il value dell'input al valore dello stato si passa praticamente da un componente uncontrolled e in cui il valore viene settato "bypassando react" ad un controlled component in cui invece ci si affida a react per gestire le modifiche al DOM.